src/os/bluestore/BlueStore.h

```diff
@@ -513,8 +513,9 @@ class BlueStore : public ObjectStore,
       ++nref;
     }
     void put() {
-      if (--nref == 0)
-	delete this;
+      if (--nref == 0) {
+	//delete this;
+      }
```

Comment:
Sorry for confusion. Speaking of 'dirty' blobs I didn't mean anything related to dirty_blob() func. It was just another aspect of my point above - reallocating object with unpredictable internal state (inherited from the initial usage) is inappropriate. Caller most probably expects some initial state...
Hence we need to do the cleanup on putting into the pool for two purposes:
1) do not waste memory if possible
2) have a predictable(initial) object state on second+ allocation.